pkg/obfuscator/ip.go

```diff
@@ -80,8 +74,8 @@ func NewIPObfuscator(replacementType schema.ObfuscateReplacementType) (Obfuscato
 	return &ipObfuscator{
 		ReplacementTracker: NewSimpleTracker(),
 		replacements: map[*regexp.Regexp]*generator{
-			ipv4Pattern: {template: consistentIPv4Template, static: obfuscatedStaticIPv4},
-			ipv6Pattern: {template: consistentIPv6Template, static: obfuscatedStaticIPv6},
+			ipv4Pattern: {template: consistentIPv4Template, static: obfuscatedStaticIPv4, replacementType: replacementType},
```

Comment:
can't comment on the above line, but you can remove the `if replacementType != ...` checks now